Add Solara integration to the third party tools (#72)
mistralai/cookbook: third_party/solara/README.md

    text_embeddings = np.array([get_text_embedding(chunk) for chunk in chunks])
    d = text_embeddings.shape[1]
    index = faiss.IndexFlatL2(d)
    index.add(text_embeddings)

    question_embeddings = np.array([get_text_embedding(question)])
    D, I = index.search(question_embeddings, k = 3)
    retrieved_chunk = [chunks[i] for i in I.tolist()[0]]
    text_retrieved = "\n\n".join(retrieved_chunk)
    return text_retrieved

class MessageDict(TypedDict):
    role: str
    content: str

messages: sl.Reactive[List[MessageDict]] = sl.reactive([])

def response_generator(messages: list, txt: List[str]):
    response = client.chat_stream(
        model = "open-mistral-7b", 
        messages = messages[:-1] + [{"role":"user","content": rag_pdf(txt, messages[-1]["content"]) + "\n\n" + messages[-1]["content"]}],
        max_tokens = 1024
    )
    for chunk in response:
        yield chunk.choices[0].delta.content

def add_chunk_to_ai_message(chunk: str):
    messages.value = [
        *messages.value[:-1],
        {
            "role": "assistant",
            "content": messages.value[-1]["content"] + chunk,
        },
    ]

@sl.component
def Page():
    with sl.Sidebar():

>         content, set_content = sl.use_state(cast(List[bytes], []))

Reviewer: ```suggestion
```
Not needed anymore, with the get_text task.
Also, I would not mix use_state and use_reactive, I would simply use one (I prefer use_reactive), it will only confuse new users I think.

>         filename, set_filename = sl.use_state(cast(List[str], []))

Reviewer: ```suggestion
        filename, set_filename = sl.use_state(cast(List[str], []))
```
not used

>         txt = sl.use_reactive(cast(List[str], []))

Reviewer: ```suggestion
```
No need for an extra reactive variable, the less state the better, we can use the get_txt.value now

>         def on_file(files: List[FileInfo]):
>             set_filename([file["name"] for file in files])
>             set_content([file["file_obj"].read() for file in files])

Reviewer: ```suggestion
        def on_file(files: List[FileInfo]):
            get_text([file["data"] for file in files])
```

>         def get_text():
>             txt_all = []
>             for _content in content:
>                 bytes_io = io.BytesIO(_content)
>                 reader = PyPDF2.PdfReader(bytes_io)
>                 txt_aux = ""
>                 for page in reader.pages:
>                     txt_aux += page.extract_text()
>                 txt_all.append(txt_aux)
>             return txt_all

Reviewer: ```suggestion
        @solara.task
        def get_text(pdf_content):
            txt_all = []
            for _content in pdf_content:
                bytes_io = io.BytesIO(_content)
                reader = PyPDF2.PdfReader(bytes_io)
                txt_aux = ""
                for page in reader.pages:
                    txt_aux += page.extract_text()
                txt_all.append(txt_aux)
            return txt_all
```
I would create a top level task, and call that directly from `on_file`. That reduces complexity instead of executing the task indirectly via the content state, which is not used anymore.

Author: This is very nice. I love this @solara.task

        sl.FileDropMultiple(
            label="Drag and drop your PDF file(s) here.",
            on_file=on_file,
            lazy=True,
        )

>         if content:
>             sl.Info("File(s) has been uploaded. Showing the beginning of the file(s)...")
>             result: Task[List[str]] = use_task(get_text, dependencies=[content])
>             if result.finished:
>                 txt.value = result.value
>             sl.ProgressLinear(result.pending)
>             for text in txt.value:
>                 sl.Markdown(f"{text[:100]}")

Reviewer: ```suggestion
        if content:
            sl.Info("File(s) has been uploaded. Showing the beginning of the file(s)...")
            result: Task[List[str]] = use_task(get_text, dependencies=[content])
            if result.finished:
                txt.value = result.value
            sl.ProgressLinear(result.pending)
            for text in txt.value:
                sl.Markdown(f"{text[:100]}")
```
```suggestion
        sl.ProgressLinear(get_text.pending)
        if get_text.value:
            sl.Info("File(s) has been uploaded. Showing the beginning of the file(s)...")
            for text in get_text.value:
                sl.Markdown(f"{text[:100]}")
```

Reviewer: The conditional hook is now allowed; using a top-level task can avoid this.
